platform-service/src/unstract/platform_service/main.py

```diff
@@ -182,6 +184,61 @@ def health_check() -> str:
     return "OK"
 
 
+@app.route("/page-usage", methods=["POST"], endpoint="page_usage")
+@authentication_middleware
+def page_usage() -> Any:
+    """Usage endpoint."""
+    result: dict[str, Any] = {
+        "status": "ERROR",
+        "error": "",
+        "unique_id": "",
+    }
+    payload: Optional[dict[Any, Any]] = request.json
+    if not payload:
+        result["error"] = INVALID_PAYLOAD
+        return make_response(result, 400)
+
+    bearer_token = get_token_from_auth_header(request)
+    _, org_id = get_organization_from_bearer_token(bearer_token)
+
+    page_count = payload.get("page_count", "")
+    file_name = payload.get("file_name", "")
+    file_size = payload.get("file_size", "")
+    file_type = payload.get("file_type", "")
+    run_id = payload.get("run_id", "")
+
+    query = f"""
+            INSERT INTO {DBTable.PAGE_USAGE} (id, organization_id, pages_processed,
+            file_name, file_size, file_type, run_id, created_at)
+            VALUES (%s, %s, %s, %s, %s, %s, %s, %s)
+        """
+    usage_id = uuid.uuid4()
+    current_time = datetime.now()
+    params = (
+        usage_id,
+        org_id,
+        page_count,
+        file_name,
+        file_size,
+        file_type,
+        run_id,
+        current_time,
+    )
+
+    try:
+        with be_db.atomic() as transaction:
```

Comment:
@johnyrahul Trying to understand this... Why do we need an atomic block here? I see only one Insert query being executed here... Is that right?

Reply:
@gaya3-zipstack Yes, you are correct, I followed the same as what we are doing for the token usage endpoint. I don't know if that is added for any specific reason. @Deepak-Kesavan  any idea?

Reply:
Anyways it won't do any damage, I believe.